src/client/Microsoft.Identity.Client/PlatformsCommon/Shared/InMemoryTokenCacheAccessor.cs

```diff
@@ -154,62 +156,27 @@ public void DeleteAccount(MsalAccountCacheKey cacheKey)
         #region Get All Values
         public IEnumerable<MsalAccessTokenCacheItem> GetAllAccessTokens()
         {
-            // perf: do not call ConcurrentDictionary.Values as it takes a lock
-            List<MsalAccessTokenCacheItem> ats = new List<MsalAccessTokenCacheItem>();
-            foreach (var kvp in _accessTokenCacheDictionary)
-            {
-                ats.Add(kvp.Value);
-            }
-
-            return ats;
+            return _accessTokenCacheDictionary.Select(kv => kv.Value);
         }
 
         public IEnumerable<MsalRefreshTokenCacheItem> GetAllRefreshTokens()
         {
-            // perf: do not call ConcurrentDictionary.Values as it takes a lock
-            List<MsalRefreshTokenCacheItem> rts = new List<MsalRefreshTokenCacheItem>();
-            foreach (var kvp in _refreshTokenCacheDictionary)
-            {
-                rts.Add(kvp.Value);
-            }
-
-            return rts;
+            return _refreshTokenCacheDictionary.Select(kv => kv.Value).ToList();
```

Comment:
@bgavrilMS As discussed, added `ToList` in these methods, since we need to investigate what the impact of returning an `IEnumerable` is.